### DISCUSSION TITLE
Redirect on login using default auth js sign in page

### DISCUSSION THREAD

**Discussion by Ivor808** [nextauthjs/next-auth#9658] [Help]
Hey all! I'm using next auth with next js and I'm having trouble redirecting users to not my root url. I'd like them on successful login to be redirected to say /home rather than just /. How can this be accomplished? I know that the client api has access to `callbackUrl`, is there something similar we can use with the default sign in flow? Thank you!

**Comment by ivangreene** [comment]
It seems like the 'callbackUrl' is somewhat overloaded in meaning. It is at least used for:
1. Sending to the external auth provider to callback redirect with the external token (callback url)
2. Redirecting from an auth action request handler

So sometimes it is set to the root of the app: https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/lib/utils/callback-url.ts#L21, and sometimes it is set to for example `/api/auth/callback/github`

Trying to sort this out as well to get better behavior in my app
